store/proof.go

```diff
@@ -62,38 +54,6 @@ func NewSMTCommitmentOp(key []byte, proof *ics23.CommitmentProof) CommitmentOp {
 	}
 }
 
-// CommitmentOpDecoder takes a ProofOp and attempts to decode it into a CommitmentOp
-// ProofOperator. The proofOp.Data is just a marshaled CommitmentProof. The Key
-// of the CommitmentOp is extracted from the unmarshalled proof.
-func CommitmentOpDecoder(pop cmtcrypto.ProofOp) (merkle.ProofOperator, error) {
```

Comment:
But it relies on `cmtcrypto.ProofOp` as an argument. What exactly do you propose?

Reply:
need to open the code editor, i was playing with this code before to avoid the comet dependency but still be able to prove. Seems like the code you added does that anyways.